lib/internal/Magento/Framework/Data/Form/Filter/Trim.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+/**
+ * Form Input/Output Trim Filter
+ *
+ * @author      Magento Core Team <core@magentocommerce.com>
+ */
+namespace Magento\Framework\Data\Form\Filter;
+
+class Trim implements \Magento\Framework\Data\Form\Filter\FilterInterface
+{
+    /**
+     * Returns the result of filtering $value
+     *
+     * @param string $value
+     * @return string
+     */
+    public function inputFilter($value)
+    {
+        return trim($value, ' ');
```

Comment:
Did you find a reason to not remove other whitespaces as done by `trim()` by default?

Reply:
I could be wrong of course, but I indeed filtered ordinary spaces only on purpose, to make sure the trim-function remains strict (matching the expected behaviour stated within the original ticket) and to avoiding unwanted removal of other characters (for example, when text area inputs should accept and store newlines, but redundant whitespaces dropped).

Other than that, no real reason to deviate from the `trim()` default behaviour indeed :) Let me know if you'd prefer the default behaviour after all, I'll gladly correct the PR.

Reply:
Thanks for your explanation! Let's keep it like that for now. If demand shows we should trim for other whitespaces, this still can be done in the future.